packages/material-ui-lab/src/Timeline/Timeline.tsx

```diff
@@ -48,20 +40,75 @@ export interface TimelineProps extends StandardProps<React.HTMLAttributes<HTMLUL
     /** Styles applied to the root element if `align="alternate"`. */
     alignAlternate?: string;
   };
+
+  /**
+   * className applied to the root element.
+   */
+  className?: string;
+  /**
+   * The system prop that allows defining system overrides as well as additional CSS styles.
+   */
+  sx?: SxProps<Theme>;
 }
 
-const Timeline = React.forwardRef<HTMLUListElement, TimelineProps>(function Timeline(props, ref) {
-  const { align = 'left', classes, className, ...other } = props;
+type Styles = Record<TimelineClassKey, any>;
+type StyleProps = TimelineProps;
+
+const useUtilityClasses = (styleProps: StyleProps) => {
+  const { align, classes } = styleProps;
+
+  const slots = {
+    root: ['root', align && `align${capitalize(align)}`],
+  };
+
+  return composeClasses(slots, getTimelineUtilityClass, classes);
+};
+
+const TimelineRoot = experimentalStyled(
+  'ul' as const,
+  {},
+  {
+    name: 'MuiTimeline',
+    slot: 'Root',
+    // @ts-ignore TODO should fix the type in experimentalStyled
```

Comment:
It's great that we can spot issues like this by actually using it in TS. Let me know if you need some help with this. Probably these can be inferred from the props' props, but also allow generic, as I think sometimes we have additional or calculated props here.

Reply:
@mnajdova I fix as workaround by changing from `string | object` to `Record<string, any>`